istioctl/cmd/waypoint.go

```diff
@@ -145,6 +151,84 @@ func waypointCmd() *cobra.Command {
 			return nil
 		},
 	}
+	waypointListCmd := &cobra.Command{
+		Use:   "list",
+		Short: "List managed waypoint configurations",
+		Long:  "List managed waypoint configurations in the cluster created by istioctl",
```

Comment:
I'm not sure the `created by istioctl` part is accurate. Since it verifies that the Gateway just conforms to expected naming conventions?  Any reason we wouldn't want all waypoints listed, regardless of whether they were created by istioctl?

Reply:
> Any reason we wouldn't want all waypoints listed, regardless of whether they were created by istioctl?

Yes I think we can list all. Also, I think we can extend the delete command to support deleting by name. My intention for creating this command was to simplify the process of deleting a specific waypoint. Currently, we have to list all gateways to find it. With this new feature, we can list the waypoints and then delete the desired one using its name or service account.

Reply:
I mean we can also include those not conforming the naming conventions but are also waypoints. WDYT?